azure-devops/azext_devops/test/pipelines/test_pipelines_build.py

```diff
@@ -0,0 +1,95 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import unittest
+
+try:
+    # Attempt to load mock (works on Python 3.3 and above)
+    from unittest.mock import patch
+except ImportError:
+    # Attempt to load mock (works on Python version below 3.3)
+    from mock import patch
+
+from azext_devops.dev.pipelines.build import (build_show)
+from azext_devops.dev.common.services import clear_connection_cache
+from vsts.build.v4_0.build_client import BuildClient
+
+class TestPipelinesBuildMethods(unittest.TestCase):
+
+    _TEST_DEVOPS_ORGANIZATION = 'https://someorganization.visualstudio.com'
+    _TEST_PAT_TOKEN = 'Some_PAT_Token'
+    
+    def setUp(self):
+
+        self.get_client = patch('vsts.vss_connection.VssConnection.get_client')
+        self.get_build_patcher = patch('vsts.build.v4_0.build_client.BuildClient.get_build')
+        self.get_credential_patcher = patch('azext_devops.dev.common.services.get_credential')
+        self.validate_token_patcher = patch('azext_devops.dev.common.services.validate_token_for_instance')
+        self.open_in_browser_patcher = patch('azext_devops.dev.pipelines.build._open_build')
+
+        #start the patchers
+        self.mock_get_build= self.get_build_patcher.start()
+        self.mock_get_client = self.get_client.start()
+        self.mock_get_credential = self.get_credential_patcher.start()
+        self.mock_validate_token = self.validate_token_patcher.start()
+        self.mock_open_browser = self.open_in_browser_patcher.start()
+
+        # Set return values which will be same across tests
+        self.mock_get_client.return_value = BuildClient(base_url=self._TEST_DEVOPS_ORGANIZATION)
```

Comment:
@ishitam8 @gauravsaralMs  Just FYI... If we mock the get_client there will be no network call.